core/tee/tee_svc_storage.c

```diff
@@ -598,7 +645,15 @@ TEE_Result tee_svc_storage_obj_create(uint32_t storage_id, void *object_id,
 		goto exit;
 	}
 
+clean:
+	tee_file_ops.unlink(tmpfile);
```

Comment:
If we enter this line not because of a `goto clean` but in the normal function flow we'll try to unlink a file that we know doesn't exist.